### PR TITLE
fix: list-app add fast parser option

### LINKED ISSUE #17: ListDownStreamFiles does not support deleted files in changed files list
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

use the test data in ```pkg/tools/list/testdata/complicate/appops/projectD``` as an example: 

set the ListDepsOption.ChangedPaths and contains a deleted file path in it, then call ListDownStreamFiles API, for example: 
  ```
  depParser, err := NewImportDepParser("pkg/tools/list/testdata/complicate/", DepOption{Files: []string{"appops/projectD/base/base.k", "appops/projectD/dev/main.k"}, ChangedPaths: []string{"base/frontend/not_exist.k"}})
  affected := depParser.ListDownStreamFiles()
  ```

### 2. What did you expect to see? (Required)

down stream files should be:
[]string{"base/frontend/not_exist.k", "base/frontend/not_exist", "appops/projectD/base/base.k", "appops/projectD/base"}

### 3. What did you see instead (Required)

the program returns file not exist error

### 4. What is your KusionStack components version? (Required)
kusion: v0.3.22-alpha.12
kclvm-go: v0.4.2-alpha2

### PR DESCRIPTION
**Pull request by chai2010**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y fix #17 #23 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [X] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [X] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Peefy — COMMENTED** [review]
Public functions and variable need more comments and examples.

File: cmds/kcl-go/command/cmd_list_app.go [hunk 1]
```diff
@@ -115,6 +121,40 @@ func NewLispAppCmd() *cli.Command {
 				}
 			}
 
+			if flagUseFastParser {
```
Comment (Peefy) [inline cmds/kcl-go/command/cmd_list_app.go:124]: add more comments on the `use-fast-parser` flag.

File: pkg/tools/list/utils.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"regex",
+	"yaml",
+	"json",
+	"crypto",
+	"base64",
+	"testing",
+	"units",
+}
+
+func isBuiltinPkg(pkgpath string) bool {
```
Comment (Peefy) [inline pkg/tools/list/utils.go:20]: add a TODO: read from kclvm rust.

File: pkg/tools/list/single_app_dep_parser.go [hunk 3]
```diff
@@ -40,41 +43,48 @@ func NewSingleAppDepParserWithFS(vfs fs.FS, opts ...Option) *SingleAppDepParser
 	return p
 }
 
-func (p *SingleAppDepParser) GetAppFiles(appPkgpath string, includeDependFiles bool) []string {
-	p.parseOnce(appPkgpath)
+func (p *SingleAppDepParser) GetAppFiles(appPkgpath string, includeDependFiles bool) ([]string, error) {
```
Comment (Peefy) [inline pkg/tools/list/single_app_dep_parser.go:46]: Add more normal and error test cases.

File: cmds/kcl-go/command/cmd_doc.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ import (
 	"kusionstack.io/kclvm-go/pkg/kclvm_runtime"
 )
 
-func NewDocCmd() *cli.Command {
+func newDocCmd() *cli.Command {
```
Comment (Peefy) [inline cmds/kcl-go/command/cmd_doc.go]: Why change these functions to private, which seems to be related to this PR

File: pkg/service/http_post.go [hunk 5]
```diff
@@ -22,9 +22,15 @@ func httpPost(urlpath string, input, output interface{}) error {
 	if err != nil {
 		return err
 	}
-	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
```
Comment (Peefy) [inline pkg/service/http_post.go:27]: Add a comment to the req Header?

File: pkg/service/rest_client_test.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package service
+
+import (
+	"testing"
+
+	"kusionstack.io/kclvm-go/pkg/spec/gpyrpc"
+)
+
+func TestCallRestMethod_ping(t *testing.T) {
```
Comment (Peefy) [inline pkg/service/rest_client_test.go:9]: Add an error test case like method not found.

File: cmds/kcl-go/command/cmd_doc.go (hunk 4 above)
Comment (chai2010) [reply cmds/kcl-go/command/cmd_doc.go]: reverted

File: pkg/tools/list/utils.go (hunk 2 above)
Comment (chai2010) [reply pkg/tools/list/utils.go:20]: fixed

File: pkg/service/http_post.go (hunk 5 above)
Comment (chai2010) [reply pkg/service/http_post.go:27]: done

File: pkg/service/rest_client_test.go (hunk 6 above)
Comment (chai2010) [reply pkg/service/rest_client_test.go:9]: fixed

File: pkg/tools/list/single_app_dep_parser.go (hunk 3 above)
Comment (chai2010) [reply pkg/tools/list/single_app_dep_parser.go:46]: done

File: cmds/kcl-go/command/cmd_list_app.go (hunk 1 above)
Comment (chai2010) [reply cmds/kcl-go/command/cmd_list_app.go:124]: update describe for the flag

**Review by Peefy — DISMISSED** [review]
LGTM

**Review by howieyuen — DISMISSED** [review]
lgtm

**Review by Peefy — APPROVED** [review]
LGTM

**Review by howieyuen — APPROVED** [review]

**Pull request merged**